### PR TITLE
Enable APIScan for VS-MEF

### PR DESCRIPTION
**Pull request by milopezc**
Enable APIScan for VS-MEF as required for compliance

### CODE REVIEW

**Comment by AArnott** [comment]
To change one file, it appears you're proposing to merge 6 commits. Is this something you plan to squash?

Have you validated that this actually works with our devdiv CI pipeline (by pushing to this repo via a topic branch and manually queuing a build)?

**Comment by milopezc** [comment]
Here's the validation build of these changes:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4698147&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=15480a3f-208f-51cb-1c92-300604e6d0ba

File: azure-pipelines/secure-development-tools.yml [hunk 1]
```diff
@@ -56,21 +47,23 @@ steps:
...
+    softwareName: 'VS-MEF'
+    softwareVersionNum: '$(Build.BuildId)'
+    isLargeApp: false
+    toolVersion: 'Latest'
+  condition: ne(variables['DisableAPIScan'], 'true')
+  env:
+    AzureServicesAuthConnectionString: runAs=App;AppId=$(ApiScanClientId);TenantId=$(ApiScanTenant);AppKey=$(ApiScanSecret)
 
  ### Create formatted HTML code analysis log file.
 - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@1
```
Comment (matteo-prosperi) [inline azure-pipelines/secure-development-tools.yml:69]: You can remove the first copy of tasks `SdtReport@1` and `PublishSecurityAnalysisLogs@2` in favor of this second one.
Add `condition: succeededOrFailed()` to these `SdtReport@1` and `PublishSecurityAnalysisLogs@2`.

Comment (milopezc) [reply azure-pipelines/secure-development-tools.yml:52]: Fixed.

**Review by matteo-prosperi — APPROVED** [review]
Make sure to squash

**Pull request merged**